core/arch/arm/plat-hikey/platform_config.h

```diff
@@ -32,19 +32,39 @@
 #define STACK_ALIGNMENT		64
 
 /* PL011 UART */
+#if defined(PLATFORM_FLAVOR_hikey)
+
+#define PL011_UART0_BASE	0xF8015000
+#define PL011_UART3_BASE	0xF7113000
 #if defined(CFG_CONSOLE_UART) && (CFG_CONSOLE_UART == 0)
-#define CONSOLE_UART_BASE       0xF8015000
+#define CONSOLE_UART_BASE	PL011_UART0_BASE
 #elif !defined(CFG_CONSOLE_UART) || (CFG_CONSOLE_UART == 3)
-#define CONSOLE_UART_BASE       0xF7113000
+#define CONSOLE_UART_BASE	PL011_UART3_BASE
+#else
+#error Unknown console UART
+#endif
+
+#elif defined(PLATFORM_FLAVOR_hikey960)
+
+#define PL011_UART5_BASE	0xFDF05000
+#define PL011_UART6_BASE	0xFFF32000
+#if defined(CFG_CONSOLE_UART) && (CFG_CONSOLE_UART == 5)
+#define CONSOLE_UART_BASE	PL011_UART5_BASE
+#elif !defined(CFG_CONSOLE_UART) || (CFG_CONSOLE_UART == 6)
```

Comment:
The only thing with the simplification is that if `CFG_CONSOLE_UART` is not defined, then it will default to uart0 for hikey instead of printing an error.